xapian-glib/xapian-mset-iterator.cc

```diff
@@ -288,16 +285,21 @@ xapian_mset_iterator_new (XapianMSet *mset)
  * xapian_mset_iterator_is_valid:
  * @iter: a #XapianMSetIterator
  *
- * Checks whether @iter is valid.
+ * Prior to 1.4, this function would check whether @iter was valid.  However
+ * #XapianMSetIterator objects are now always valid, and this function always
+ * returns %TRUE.
+ *
+ * Returns: %TRUE
  *
- * Returns: %TRUE if the iterator is valid
+ * Deprecated: 1.4: The iterator is now always valid, so checks using this
```

Comment:
I've replaced the "AVAILABLE_IN" with a "DEPRECATED_IN" like so:

```
-XAPIAN_GLIB_AVAILABLE_IN_1_0
+XAPIAN_GLIB_DEPRECATED_IN_1_6
 gboolean                xapian_mset_iterator_is_valid           (XapianMSetIterator *iter);
```

It seems really both are true still, but having both there results in a compile error:

```
error: conflicting specifiers in declaration of ‘xapian_mset_iterator_is_valid’
```

If there's a way to specify both, let me know what it is.  There aren't currently any deprecated functions I can look at for clues.

I had to rebase onto current master to get the new macros, so I also squashed in the above change plus updating both mentions of "1.4" to "1.6", and force pushed.